include/xlnt/utils/exceptions.hpp

```diff
@@ -332,16 +249,30 @@ class XLNT_API unsupported : public exception
     /// feature.
     /// </summary>
     explicit unsupported(const std::string &message);
+};
 
+/// <summary>
+/// Exception for encoding errors
+/// </summary>
+class XLNT_API encoding_error : public exception
+{
+public:
     /// <summary>
-    /// Default copy constructor.
+    /// Constructs an encoding_error exception with a message describing the encoding error.
     /// </summary>
-    unsupported(const unsupported &) = default;
+    explicit encoding_error(const std::string &message);
+};
 
+/// <summary>
+/// Exception for xlnt::variant access
+/// </summary>
+class XLNT_API bad_variant_access : public exception
+{
+public:
     /// <summary>
-    /// Destructor
+    /// Constructs a bad_variant_access exception given the mismatched types.
     /// </summary>
-    ~unsupported() override;
+    explicit bad_variant_access(int expected_type, int actual_type);
```

Comment:
Isn't it better to use `variant::type` as parameter type instead of a plain int?